### ISSUE TITLE
Styles: the edit Palette UI is not operable with a keyboard

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#62728]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/62698

I couldn't find a way to _add_ or _remove_ a custom color in the edit palette UI when using only the keyboard. 
This is a serious violation as _any_ user interface must be operable with the keyboard.

I guess this issue is not limited to the Edit palette UI and can be reproduced wherever the color picker is used with no explicit buttons to Cancel and Save.

### Step-by-step reproduction instructions

- Go to the Site Editor > Styles > Colors > Palette.
- Use the keyboard.

First not accessible flow:
- Move focus to the plus icon button 'Add color' and press Enter or Spacebar.
- The color picker opens.
- Observe that although the focus style is not always visible, all controls within the color picker aroperable with a keyboard.
- But...
- Once a color is set, there is no way to save or cancel and close the color picker with the keyboard.
- Observe there's no Button to Save.
- Press Escape: observe that nothing happens. Expected: the color value to be reset and the color picker to close.
- Use the mouse to click 'Done'.

Second not accessible flow:
- Move focus to the circular button of the new color and press Enter or Spacebar.
- The color picker opens.
- Change the values of the color usign the keyboard.
- Press Escape.
- Observe that now the color picker does close and the changes are saved.
- However, this is not the expected interaction: the Escape key should never, ever 'save'. Instead, it should always be used to cancel the current operation and reset any value.

Thir not accessible flow:
- Once a custom color is set, mofe focus to the ellipsis button 'Color options', press Enter, choose 'Show details' and press Enter.
- Move focus to the custom color item and press Enter or Spacebar.
- The color picker opens.
- Obserrve that whatever you do you can't exit the color picker with the keyboard.
- Observe there is no way to move focus to the `lineSolid` icon button to remove the color.
- Observe the remoce color button is only rendered in this 'editing' state of the UI, when keyboard users can't exit the color picker.

### Screenshots, screen recording, code snippet

For clarity, this is the UI to be tested:

![Screenshot 2024-06-21 at 11 03 28](https://github.com/WordPress/gutenberg/assets/1682452/e69c2c3b-f741-4b99-a3a5-f72f29873983)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Note: https://github.com/WordPress/gutenberg/pull/62753 is going to fix some of the issues of this UI by making it at least operable with a keyboard. There are pending issues and improvements that can be addressed here.

**Comment by afercia** [comment]
Couple important pointe to address from @joedolson's feedback on https://github.com/WordPress/gutenberg/pull/62753#issuecomment-2183555122:

> 1. Have a 'cancel' option in the color popover, so that it's possible to change a color, then return to the previous color without needing to have the hex value memorized. Right now, if you change a color, but don't have the previous value recorded, there's no way to return to the previous color.
> 2. Have the `esc` key trigger the cancel behavior, and have a button available to save the changes. I think that a click outside of the popover should be treated as a completion action; but escape should be treated as a cancellation.

**Comment by afercia** [comment]
Reopening, as https://github.com/WordPress/gutenberg/pull/62753 only fixed some of the existing issues.
